### PR TITLE
NIFI-13674 Added merge.reason attribute for MergeRecord.

### PR DESCRIPTION
**Pull request by dan-s1**
# Summary

[NIFI-13674](https://issues.apache.org/jira/browse/NIFI-13674)

I copied the description of the `WritesAttribute`  annotation from `MergeContent` `merge.reason`. Please confirm whether that is okay.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for adding this attribute to `MergeRecord` @dan-s1. For more programmatic-focused handling, I recommend creating an `enum` and using the enum name for the reason, as opposed to a string explanation. That would following the pattern of the `EvictionReason` in `MergeContent`.

**Comment by dan-s1** [comment]
Just to clarify you want method signatures of 
`RecordBinManager.handleCompletedBins(final Predicate<RecordBin> completionTest, final String completionReason)`
and
`RecordBin.complete(final String completionReason)`

to now take an enum value instead of a string?

Also the values for completionReason I see are 

1. Bin has reached Max Bin Age
2. Bin is full enough
3. Bin is completely full

Is there a way we could use the existing `EvictionReason` enum to map the above values to values in the enum (i.e. use the specific values in `EvictionReason` )?

File: nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeRecord.java [hunk 1]
```diff
@@ -171,6 +174,7 @@ public class MergeRecord extends AbstractSessionFactoryProcessor {
     public static final String MERGE_COUNT_ATTRIBUTE = "merge.count";
     public static final String MERGE_BIN_AGE_ATTRIBUTE = "merge.bin.age";
     public static final String MERGE_UUID_ATTRIBUTE = "merge.uuid";
+    public static final String REASON_FOR_MERGING = "merge.reason";
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MergeRecord.java:177]: Recommend adjusting the naming to provide some differentiation from `MergeContent`, which has a different set of defined values.
```suggestion
    public static final String MERGE_COMPLETION_REASON = "merge.completion.reason";
```

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments @dan-s1, the latest version looks good! +1 merging

**Pull request merged**